project-addons/flask_middleware_connector/events/partner_events.py

```diff
@@ -127,61 +147,102 @@ def delay_export_partner_write(session, model_name, record_id, vals):
+                for line in rma.claim_line_ids:
+                    if line.product_id.web == 'published' and \
+                            (not line.equivalent_product_id or
+                             line.equivalent_product_id.web == 'published'):
+                        export_rmaproduct.delay(session, 'claim.line', line.id,
+                                                priority=10, eta=240)
+
+        elif "web" in vals and not vals["web"]:
+            unlink_partner.delay(session, model_name, record_id, priority=100)
+            for contact in contacts:
+                export_partner.delay(session, model_name, contact.id, priority=1,
+                                     eta=120)
+        elif "active" in vals and not vals["active"] and partner.web:
+            unlink_partner(session, model_name, record_id)
+            for contact in contacts:
+                export_partner.delay(session, model_name, contact.id, priority=1,
+                                     eta=120)
+        elif 'child_ids' in vals:
+                for child in vals['child_ids']:
+                    # 2 is the state when the child is delete from the partner
+                    if 2 in child:
+                        # child estructure is [number, record_id, data] the number indicate
+                        # de status of the object and the second position is the record of the
+                        # object. The third position is the data of the object, if the object
+                        # is created is False, else if the object is creating in this moment
+                        # this position have all the data from the object and the second position
+                        # is null because the object is not created yet
+                        unlink_partner.delay(session, model_name, child[1], priority=2)
+
+        elif partner.web and (vals.get('is_company', False) or partner.is_company):
+            for field in up_fields:
+                if field in vals:
+                    update_partner.delay(session, model_name, record_id, priority=2)
+                    break
+    else:
+        if partner.parent_id.web and vals.get('active', False):
+            export_partner.delay(session, model_name, record_id, priority=1,
+                                 eta=60)
+        elif partner.parent_id.web and partner.active:
+            update_partner.delay(session, model_name, record_id, priority=3,
+                                 eta=60)
+        elif partner.parent_id.web and not vals.get('active', False):
+            unlink_partner.delay(session, model_name, record_id, priority=5,
+                                 eta=60)
 
 
 @on_record_unlink(model_names='res.partner')
 def delay_unlink_partner(session, model_name, record_id):
     partner = session.env[model_name].browse(record_id)
+
     if partner.web:
         unlink_partner.delay(session, model_name, record_id)
```

Comment:
creo que si entra por aquí también debéis de enviar el unlink a los contactos o sino controlar el constraint de Flask con un on_delete cascade

Reply:
Ten en cuenta de que la fk no es contra id, sino contra odoo_id y no se si lo hace por defecto, habría que probarlo